taco_parser/src/xml_parser.rs

```diff
@@ -0,0 +1,245 @@
+use serde::{Deserialize, Serialize};
+use serde_xml_rs;
+use std::collections::HashMap;
+use std::fs::{read, read_to_string};
+use std::path::Path as OsPath;
+
+use crate::trail_parser::{self, TrailCoordinates};
+
+#[cfg(test)]
+#[path = "xml_parser_tests.rs"]
+mod xml_parser_tests;
+
+#[derive(Debug, Deserialize)]
+pub struct OverlayData {
+    #[serde(rename = "MarkerCategory")]
+    pub marker_category: MarkerCategory,
+    #[serde(rename = "POIs")]
+    pois: POIArrayContainer,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct MarkerCategory {
+    name: String,
+    #[serde(rename = "iconFile")]
+    icon_file: Option<String>,
+    #[serde(rename = "MarkerCategory", default = "default_children")]
+    children: Vec<MarkerCategory>,
+    #[serde(rename = "heightOffset")]
+    height_offset: Option<f32>,
+}
+
+fn default_children() -> Vec<MarkerCategory> {
+    return Vec::new();
+}
+
+#[derive(Debug, Deserialize)]
+pub struct POIArrayContainer {
+    #[serde(rename = "$value")]
+    poi_array: Vec<PoiItems>,
+}
+
+#[derive(Debug, Deserialize)]
+struct POI {
+    #[serde(rename = "MapID")]
+    map_id: u32,
+    xpos: f32,
+    ypos: f32,
+    zpos: f32,
+    #[serde(rename = "iconFile")]
+    icon_file: Option<String>,
+    #[serde(rename = "heightOffset")]
+    height_offset: Option<f32>,
+    #[serde(rename = "type")]
+    type_: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct TrailMetadata {
+    #[serde(rename = "trailData")]
+    trail_data: String,
+    texture: String,
+    // color: String,
+    #[serde(rename = "iconFile")]
+    icon_file: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+enum PoiItems {
+    POI(POI),
+    #[serde(rename = "Trail")]
+    TrailMetadata(TrailMetadata),
+}
+
+#[derive(Debug, Serialize)]
+pub struct FinalResult {
+    icons: Vec<Icon>,
+    paths: Vec<Path>,
+}
+
+#[derive(Debug, Serialize, Clone)]
+struct Path {
+    texture: String,
+    points: Vec<[f32; 3]>,
+}
+
+#[derive(Debug, Serialize, Clone)]
+struct Icon {
+    position: [f32; 3],
+    texture: String,
+}
+
+//https://docs.rs/serde-xml-rs/0.5.1/serde_xml_rs/
+pub fn parse_xml(xml: &str) -> OverlayData {
+    let data: OverlayData = serde_xml_rs::from_str(xml).unwrap();
```

Comment:
I'm not going to be so strict as to say we should not have unwraps, however some of them are making errors fully opaque. I'm guessing this one is one of the culprits. If there is an error with the XML we need to show it to the user so that it can be fixed. I tried loading some of the reactif markers (https://reactif.games/taco/) which caused this module to panic, then caused burrito to crash when it tried to parse the resulting null as a json object. I think at the very minimum we need to have a `error at line #<xml_line> : <errormsg>` get bubbled up to godot so the xml files can be manually fixed if they have any errors.

Reply:
```Successfully set transient property!
New Map
Saving Old Map
Loading New Map
Loading XML file from path/home/jl/hdd_drive/GW2/taco_markers/POIs/General__TheSilverwastes.xml
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { field: "missing field `MarkerCategory`" }', src/xml_parser.rs:94:57
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
ERROR: <native>: gdnative-core: method panicked (check stderr for output)
   At: src/lib.rs:37.
SCRIPT ERROR: load_taco_markers: Invalid type in function 'parse' in base '_JSON'. Cannot convert argument 1 from Nil to String.
   At: res://Spatial.gdc:297.
```
Hmm the error message does seem to get passed properly from Rust to Godot for me when I tried one of the files from Reactif (at least in so far as appearing in the terminal).
```Custom { field: "missing field `MarkerCategory`" }', src/xml_parser.rs:94:57```

Can you share the file you were trying with, so I can reproduce? And when you say error getting bubbled up to godot, do you mean you would like it to pop up as an error in godot?

Reply:
I believe these error are being printed directly to stdout/stderr from the rust module and not being passed into the main burrito process to be printed. Thus there is no way for burrito to be able to capture them in the future to show to the user.